File: Leetcode/2094 - Finding 3-Digit Even Numbers/solution.py
class Solution:
    def findEvenNumbers(self, digits: List[int]) -> List[int]:
        # Count occurrences of each digit (0–9)
        count = [0] * 10
        for digit in digits:
            count[digit] += 1
        result = []

        # Try all 3-digit even numbers from 100 to 998
        for num in range(100, 1000, 2):  # step 2 ensures the number is even
            d1 = num // 100
            d2 = (num // 10) % 10
            d3 = num % 10

            # Make a copy of the original digit count
            temp_count = count[:]
            if temp_count[d1] > 0:
                temp_count[d1] -= 1
                if temp_count[d2] > 0:
                    temp_count[d2] -= 1
                    if temp_count[d3] > 0:
                        result.append(num)

        return result

File: Leetcode/2094 - Finding 3-Digit Even Numbers/solution.ts
function findEvenNumbers(digits: number[]): number[] {
    const count: number[] = new Array(10).fill(0);
    for (const digit of digits) {
        count[digit]++;
    }
    const result: number[] = [];

    // Try all 3-digit numbers from 100 to 998 (step by 2 to ensure even)
    for (let num = 100; num <= 999; num += 2) {
        const d1: number = Math.floor(num / 100);
        const d2: number = Math.floor((num / 10)) % 10;
        const d3: number = num % 10;

        const tempCount: number[] = [...count];
        if (tempCount[d1]-- > 0 && tempCount[d2]-- > 0 && tempCount[d3]-- > 0) {
            result.push(num);
        }
    }

    return result;
}